### PR TITLE
cmd/modelcmd: return ModelCommand from Wrap

### PR DESCRIPTION
**Pull request by rogpeppe**
This will allow commands that use modelcmd.Wrap
to inject the client store and the API open function
directly rather than relying on a custom NewXXXCommandForTest
initializer.

This hopefully means that we'll be able to have significantly less
ad hoc hackery to inject the standard mocked parts
into juju commands and more confidence that
ModelCommand commands aren't using the real
client store and/or API opening functions unless
we explicitly decide to.

Changes to the tests in a subsequent PR. No
externally visible behaviour changes in this one.

Credit to Ales Stimec for the idea.

### CODE REVIEW

File: cmd/modelcmd/modelcommand.go [hunk 1]
```diff
@@ -267,7 +267,21 @@ func Wrap(c ModelCommand, options ...WrapOption) cmd.Command {
 	for _, option := range options {
 		option(wrapper)
 	}
-	return WrapBase(wrapper)
+	// Define a new type so that we can embed the ModelCommand
+	// interface one level deeper than cmd.Command, so that
+	// we'll get the Command methods from WrapBase
+	// and all the ModelCommand methods not in cmd.Command
+	// from modelCommandWrapper.
```
Comment (mhilton) [inline cmd/modelcmd/modelcommand.go]: what's modelCommandWrapper?

**Comment by fabricematrat** [comment]
LGTM

File: cmd/modelcmd/modelcommand.go (hunk 1 above)
Comment (rogpeppe) [inline cmd/modelcmd/modelcommand.go]: It's the type we're using above that wraps c.

Comment (howbazaar) [inline cmd/modelcmd/modelcommand.go]: Seems a little magical, but the comment helps.

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**